src/ai/backend/manager/models/user.py

```diff
@@ -670,36 +671,34 @@ async def _post_func(conn: SAConnection, result: Result) -> Row:
                 else:
                     # User becomes non-admin. Make the keypair non-admin as well.
                     # If there are multiple admin keypairs, inactivate them.
+                    # TODO: Should elaborate keypair inactivation policy.
                     rows = result.fetchall()
-                    cnt = 0
                     kp_updates = []
-                    for row in rows:
+                    for idx, row in enumerate(rows):
                         kp_data = {
-                            "user": row.user,
-                            "is_admin": keypairs.c.is_admin,
-                            "is_active": keypairs.c.is_active,
+                            "b_access_key": row.access_key,
+                            "is_admin": row.is_admin,
+                            "is_active": row.is_active,
                         }
-                        changed = False
-                        if cnt == 0:
+                        if idx == 0:
                             kp_data["is_admin"] = False
-                            changed = True
-                        elif row.is_admin and row.is_active:
+                            kp_updates.append(kp_data)
+                            continue
+                        if row.is_admin and row.is_active:
                             kp_data["is_active"] = False
-                            changed = True
-                        if changed:
                             kp_updates.append(kp_data)
-                        cnt += 1
-                    await conn.execute(
-                        sa.update(keypairs)
-                        .values(
-                            {
-                                "is_admin": bindparam("is_admin"),
-                                "is_active": bindparam("is_active"),
-                            }
+                    if kp_updates:
```

Comment:
Oops! Same here. 
```suggestion
                    if len(kp_updates) > 0:
```

Reply:
I already updated this one, but how about using Python truthness?